### PR TITLE
[BEAM-1194] Add DataflowLocationIT

### PR DESCRIPTION
**Pull request by dhalperi**
Opening PR to test.

### CODE REVIEW

**Comment by dhalperi** [comment]
Passing test run: https://builds.apache.org/view/Beam/job/beam_PostCommit_Java_MavenInstall/2199/org.apache.beam$beam-runners-google-cloud-dataflow-java/testReport/org.apache.beam.runners.dataflow/DataflowLocationIT/

**Comment by dhalperi** [comment]
R: @tgroh 

Be skeptical.

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowLocationIT.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    TestDataflowPipelineOptions options =
+        TestPipeline.testingPipelineOptions().as(TestDataflowPipelineOptions.class);
+    Pipeline p = Pipeline.create(options);
+
+    options.setTempLocation(ensureNoSlash(options.getTempLocation()));
+    options.setGcpTempLocation(ensureNoSlash(options.getGcpTempLocation()));
+    options.setStagingLocation(ensureNoSlash(options.getStagingLocation()));
+    runTestPipeline(p);
+
+    // Assert that the options were not changed, at least visibly.
```
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowLocationIT.java]: This is probably too strict a check - they should point at the same location, but need not be equivalent literals.

I've filed https://issues.apache.org/jira/browse/BEAM-1210 - which will cause these assertions to not be particularly meaningful

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowLocationIT.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+/**
+ * Tests that the Dataflow runner accepts various --XLocation options with and without trailing '/'.
+ */
+@RunWith(JUnit4.class)
+public class DataflowLocationIT {
+
+  /**
+   * Builds and runs the test using the given {@link Pipeline} We have to invoke the code in this
+   * awkward way so that the calling test can set options after the {@link TestDataflowRunner} has
```
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowLocationIT.java]: This seems like a relatively straightforward pipeline. I'm not sure what awkward way you're referring to.

**Comment by dhalperi** [comment]
Closing this in favor of a different approach.

**Pull request closed without merging**